### PR TITLE
Adds the watch-plugin make target to improve the development experience

### PR DESCRIPTION
**Pull request by mgdelacroix**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -101,6 +101,9 @@ webapp: ## Build webapp.
 watch-webapp: ## Run webapp watching for changes.
 	cd webapp; npm run watchdev
 
+watch-plugin: ## Run and upload the plugin to a development server
```
Comment (wiggin77) [inline Makefile:104]: Can we check that modd is installed, and install it?  Or at least a note here saying how to do it?

Comment (mgdelacroix) [reply Makefile]: Sure! Fixed in fd57d81

File: Makefile [hunk 2]
```diff
@@ -89,10 +89,16 @@ server-lint: ## Run linters on server code.
 server-test: ## Run server tests
 	cd server; go test -race -v -count=1 ./...
 
-watch-server: ## Run server watching for changes with modd (https://github.com/cortesi/modd).
+modd-precheck:
+	@if ! [ -x "$$(command -v modd)" ]; then \
+		echo "modd is not installed. Please see https://github.com/cortesi/modd#install for installation instructions"; \
```
Comment (wiggin77) [inline Makefile:94]: Perfect!

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Review by jespino — APPROVED** [review]
LGTM

**Pull request merged**